harness/determined/_trial_controller.py

```diff
@@ -41,13 +42,21 @@ def __init__(
         load_path: Optional[pathlib.Path],
         rendezvous_info: RendezvousInfo,
         hvd_config: horovod.HorovodContext,
+        prof: profiler.ProfilerAgent,
     ) -> None:
         self.context = context
         self.env = env
         self.workloads = workloads
         self.load_path = load_path
         self.rendezvous_info = rendezvous_info
         self.hvd_config = hvd_config
+        self.prof = prof
+
+        self.prof = profiler.ProfilerAgent.from_env(
+            env,
+            rendezvous_info.get_rank(),
+            int(os.getenv("HOROVOD_LOCAL_RANK", 0)),
```

Comment:
afaict this is the same env where we were getting it before so it just be there/just fine

Reply:
please do not read `HOROVOD_LOCAL_RANK`, use `context.distributed.get_rank()` instead.